### PR TITLE
enh: set_allocation_size for transpose intrinsic array function

### PR DESCRIPTION
**Pull request by gxyd**
## Description

I'll first want to see if the tests pass or not. The idea is to make the test case being added here: https://github.com/lfortran/lfortran/pull/5349 to pass with `--experimental-simplifier`.

Currently, I've commented out the test case because it doesn't work without *experimental-simplifier*.

### CODE REVIEW

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -930,6 +930,7 @@ RUN(NAME intrinsics_322 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) #
 RUN(NAME intrinsics_323 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc) # dot_product
 RUN(NAME intrinsics_324 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # max
 RUN(NAME intrinsics_325 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # spread
+# RUN(NAME intrinsics_326 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc fortran) # transpose
```
Comment (gxyd) [inline integration_tests/CMakeLists.txt:933]: currently there is no label which allows to run a test with only *experimental-simplifier* (i.e. not run the test case unless *simplifier* pass is used)

Comment (gxyd) [reply integration_tests/CMakeLists.txt:933]: https://github.com/lfortran/lfortran/pull/5349 fixes this issue without use of *simplifier pass*.

Comment (certik) [reply integration_tests/CMakeLists.txt:933]: go ahead and rebase this PR and enable the test.

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -523,7 +523,7 @@ RUN(NAME arrays_55 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 # is enabled i.e. works on `simplifier_pass` branch
 # RUN(NAME arrays_56 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
 RUN(NAME arrays_57 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
-RUN(NAME arrays_58 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc NO_EXPERIMENTAL_SIMPLIFIER)
+RUN(NAME arrays_58 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
```
Comment (gxyd) [inline integration_tests/CMakeLists.txt:526]: the fix in this PR, enables this with `--experimental-simplifier` flag

**Pull request merged**